### DISCUSSION TITLE
skynet 适配 mongodb 哪个版本

### DISCUSSION THREAD

**Discussion by opaser** [cloudwu/skynet#1634] [General]
mongodb 6.0 无法使用

**Comment by cloudfreexiao** [comment]
最高 mongo 5.0

https://www.mongodb.com/docs/manual/reference/mongodb-wire-protocol/#request-opcodes

**Comment by cloudwu** [comment]
skynet 不关联任何 mongo ，是 mongo driver 有限制。 mongo driver 是一个独立项目，不只用在 skynet 中。

参考 https://github.com/cloudwu/skynet/discussions/1619

等有人改进 mongo driver ，支持 `OP_MSG` 。

事情不难做，就是按新协议打包 OP_MSG 取代就有的那几条。协议在上面的链接中有提到。
连上 db 后检测版本，区分用旧协议还是新协议即可。

我最近忙，没空搞。

**Comment by cloudwu** [comment]
https://github.com/cloudwu/skynet/pull/1649 已支持到 6.x 。
